### ISSUE TITLE
.latex directory optional

### ISSUE THREAD

**Issue by stefan-balke** [andrewrynhard/atom-latex-plus#49]
Hey @andrewrynhard,

great package, but is there a way to deactivate the linking of the pdf file and the .latex folder?
I work with other people in a team and sometimes they are confused about those folders since their tex-tools don't use such things...

Thanks

**Comment by andrewrynhard** [comment]
Hello and thank you. Currently there is not a way do deactivate linking. Feel free to make a PR. I am also considering taking a look at removing linking all together. The idea behind the .latex files is that it would be `.gitignore`able. Having links acutally doesn't make a whole lot of sense. Keeping the output does. Would this work for your case?

**Comment by stefan-balke** [comment]
Hey,

jep, would work!
It would be great if one could set a standard foldername for the output. I would rather to prefer to not hide the output folder, so would call it `latex` instead of `.latex`.

Thanks!

**Comment by dpo** [comment]
FWIW I'm not a fan of a folder named `latex` as I already have a number of such folders, unrelated to LaTeX-plus, lying around. I imagine others do too. I'd be more in favor of placing the temp files in a temp folder (such as created by `mktemp`), but that makes looking at the logs inconvenient (in case something goes wrong).

Another difficulty is that `latexmk` doesn't allow placing the temp files in a location and the output files (e.g., the pdf) in another.

**Comment by andrewrynhard** [comment]
@stefan-balke @dpo there is a way to change the output location. Open the project settings and change the `output` entry.

**Comment by svkurowski** [comment]
Just to add my opinion:
I'm in heavy favour of building in a subdirectory and symlinking the files. Out-of-source builds are common in all programming languages (yes I know that Latex is not a programming language per se but the point still stands). It allows me put all output files into the `.gitignore` easily:

``` bash
.build # Changed from the default .latex
main.pdf
main.synctex.gz
```

A temporary folder does not work as it would require rebuilding after every restart.

So if you change something about the current behaviour please make it optional!

**Comment by stefan-balke** [comment]
> So if you change something about the current behaviour please make it optional!

Yeah, optional would be great.

> there is a way to change the output location. Open the project settings and change the output entry.

Yup, I know, but I don't want a project file for each of my little tex-projects...for a thesis or a paper this is fine...

**Comment by DeusProx** [comment]
> Yup, I know, but I don't want a project file for each of my little tex-projects...for a thesis or a paper this is fine...

ATM you get such a file for every project you compile with latex-plus. But the baddest part is that those configs are located in ~/.atom/packages/latex-plus/projects/........   so you can't even easily share them with a source code management system like github. In my opinion that's not good.

**Comment by andrewrynhard** [comment]
@svkurowski @stefan-balke The current behavior will not change. But I do think we can make everyone happy here. The `output` option can changed from a per-project setting to a global package setting. So instead of changing `.latex` to `.build` for every project you only need to change it in the package settings.

**Comment by andrewrynhard** [comment]
@DeusProx this behavior was intentional because I thought people would not want the litter their repository with this file. We can easily change this and move it to the repository root if that is what the users want.

**Comment by DeusProx** [comment]
@andrewrynhard I can only speak for myself. i would like to have this file in the root of the project. one could also check if this file exists to check if this project should be able to run compile... this would be related to one of the other issues

But I could see that not everyone wants to have this behaviour.

**Comment by andrewrynhard** [comment]
Closed by #71
